src/Clapi/Relay.hs

```diff
@@ -100,3 +102,37 @@ import Data.Maybe.Clapi (note)
 -- processOms u oms =
 --   do
 --     undefined
+
+applyMessage :: (MonadFail m) => Valuespace Unvalidated -> Message -> m (Valuespace Unvalidated)
+applyMessage vs msg = case msg of
+    (MsgSet p t v i a s) -> vsSet a i v p s t vs
+    (MsgAdd p t v i a s) -> vsAdd a i v p s t vs
+    (MsgRemove p t a s) -> vsRemove a p s t vs
+    (MsgClear p t a s) -> vsClear a p s t vs
+    (MsgAssignType p tp) -> return $ vsAssignType p tp vs
+    (MsgDelete p) -> vsDelete p vs
+    (MsgChildren p c) -> undefined -- FIXME: backing implementation
+    (MsgSubscribe _) -> return vs
+    (MsgUnsubscribe _) -> return vs
+    (MsgError _ _) -> return vs
+
+applyMessages :: Valuespace v -> [Message] -> MonadErrorMap (Valuespace Unvalidated)
+applyMessages mvvs = applySuccessful [] (unvalidate mvvs)
+  where
+    applySuccessful errs vs [] = (Map.fromList errs, vs)
+    applySuccessful errs vs (m:ms) = case canFailApply vs m of
+        (Left es) -> applySuccessful ((msgPath' m, es):errs) vs ms
+        (Right vs') -> applySuccessful errs vs' ms
+    canFailApply :: Valuespace Unvalidated -> Message -> CanFail (Valuespace Unvalidated)
+    canFailApply = applyMessage
+
+handleOwnerMessages :: Valuespace Validated -> [Message] -> ([Message], Valuespace Validated)
+handleOwnerMessages vs msgs = (rmsgs, rvs)
+  where
+    rvs = if errored then vs else vvs
+    errored = errs == Map.empty
+    errs = Map.union aErrs vErrs
```

Comment:
Unioning these maps will throw away errors on the RHS if there are errors on the same path. Is that what you meant? Are these maps of _lists_ of errors? If so, the `Mol` stuff I wrote deals with appending inside the `Map`...

Reply:
The intention is to take the first error on the path I think. Whether that's a good idea is a whole other question.